src/strophe.stream-management.js

```diff
@@ -133,17 +149,58 @@ Strophe.addConnectionPlugin('streamManagement', {
 	},
 
 	init: function(conn) {
-		console.info('INIT CONN MANAGEMENT', conn);
 		this._c = conn;
 		Strophe.addNamespace('SM', this._NS);
 
 		// Storing original xmlOutput function to use additional logic
 		this._originalXMLOutput = this._c.xmlOutput;
 		this._c.xmlOutput = this.xmlOutput.bind(this);
+
+		this._originalConnect = this._c.connect;
+		this._c.connect = this._interceptConnectArgs.bind(this);
+
+		this._originalDoDisconnect = this._c._doDisconnect;
+		this._c._doDisconnect = this._interceptDoDisconnect.bind(this);
+
+		this._originalDisconnect = this._c.disconnect;
+		this._c.disconnect = this._interceptDisconnect.bind(this);
```

Comment:
I'm thinking the connection could be a new class or passed into a decorator, so a lot of this logic can be moved out of stream-management and into a class dedicated to adding resume ability to connection. What do you think?